Crate fails to compile

### Steps to reproduce the bug with the above code

The 0.17.3 version of the `boa_icu_provider` crate fails to compile:
https://github.com/boa-dev/boa/tree/v0.17.3/boa_icu_provider

So:
 1. Clone the boa repo
 2. Checkout the v0.17.3 tag
 3. `cd boa_icu_provider`
 4. `cargo semver-checks`

With the latest `cargo-semver-checks` version in crates.io

Behaviour has also been observed in the CI/CD in Ubuntu Linux:
https://github.com/boa-dev/boa/pull/3173

### Actual Behaviour

It gives a bunch of errors, such as:
```
error[E0433]: failed to resolve: could not find `locid` in `_internal`
  --> ~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/icu_decimal-1.2.0/src/provider.rs:84:1
   |
84 | / #[icu_provider::data_struct(marker(
85 | |     DecimalSymbolsV1Marker,
86 | |     "decimal/symbols@1",
87 | |     extension_key = "nu"
88 | | ))]
   | |___^ could not find `locid` in `_internal`
   |
   = note: this error originates in the attribute macro `icu_provider::data_struct` (in Nightly builds, run with -Z macro-backtrace for more info)
```


### Expected Behaviour

Compilation works, since `cargo build`, `cargo test` and `cargo doc` all work fine.

### Generated System Information

#### Software version

cargo-semver-checks 0.24.1

#### Operating system

macOS 14.0 (Darwin 23.0.0)

#### Command-line

```bash
~/.cargo/bin/cargo-semver-checks semver-checks --bugreport 
```

#### cargo version

```
> cargo -V 
cargo 1.73.0 (9c4383fb5 2023-08-26)
```

#### Compile time information

- Profile: release
- Target triple: aarch64-apple-darwin
- Family: unix
- OS: macos
- Architecture: aarch64
- Pointer width: 64
- Endian: little
- CPU features: aes,crc,dit,dotprod,dpb,dpb2,fcma,fhm,flagm,fp16,frintts,jsconv,lor,lse,neon,paca,pacg,pan,pmuv3,ras,rcpc,rcpc2,rdm,sb,sha2,sha3,ssbs,vh
- Host: aarch64-apple-darwin

### Build Configuration

```toml
[target.x86_64-apple-darwin]
rustflags = ["-C", "link-arg=-fuse-ld=/usr/local/bin/zld"]
```

### Additional Context

It seems the error only happens with that particular crate in the workspace, and seems that the 0.18.0 (current `semver-checks` branch) works fine.

--- a/src/rustdoc_cmd.rs
+++ b/src/rustdoc_cmd.rs
@@ -178,13 +178,29 @@ impl RustdocCommand {
                 )?;
                 writeln!(
                     stderr,
-                    "note: running the following command on the crate should reproduce the error:"
+                    "note: the following command can be used to reproduce the compilation error:"
                 )?;
-
+                let selector = match crate_source {
+                    CrateSource::Registry { version, .. } => format!("{crate_name}@={version}"),
+                    CrateSource::ManifestPath { manifest } => format!(
+                        "--path {}",
+                        manifest
+                            .path
+                            .parent()
+                            .expect("source Cargo.toml had no parent path")
+                            .to_str()
+                            .expect("failed to create path string")
+                    ),
+                };
+                let feature_list = features.into_iter().join(",");
                 writeln!(
                     stderr,
-                    "      cargo build --no-default-features --features {}\n",
-                    features.into_iter().join(","),
+                    "      \
+cargo new --lib example &&
+          cd example &&
+          echo '[workspace]' >> Cargo.toml &&
+          cargo add {selector} --no-default-features --features {feature_list} &&
+          cargo check\n"
                 )?;
                 Ok(())
             })?;

